pintracker/stateless/stateless.go

```diff
@@ -216,6 +229,13 @@ func (spt *Tracker) Shutdown(ctx context.Context) error {
 	return nil
 }
 
+// SetState sets the pin state and returns true if state was not empty or nil.
+func (spt *Tracker) SetState(st state.ReadOnly) bool {
```

Comment:
Why don' t you initialize the pintracker with a new function parameter:

`New(..., func() (state.ReadOnly, error)...)`

This function can block until consensus is ready, it can either be used to request GetState() from the consensus layer all the time or just once and keep it. i.e.:

```
getState := func() (state.ReadOnly, error) {
   <-cons.Ready()
   return cons.GetState()
}
```

Reply:
I have used getState to request GetState() from the consensus layer  all the time